nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java

```diff
@@ -35,15 +35,82 @@
 
 public class ITDeleteAzureDataLakeStorage extends AbstractAzureDataLakeStorageIT {
 
+    private static final boolean FILE = true;
+    private static final boolean DIRECTORY = false;
+
     @Override
     protected Class<? extends Processor> getProcessorClass() {
         return DeleteAzureDataLakeStorage.class;
     }
 
+    @Test
+    public void testDeleteDirectoryWithFiles() {
+        // GIVEN
+        String directory = "TestDirectory";
+        String filename = "testFile.txt";
+        String fileContent = "AzureFileContent";
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        createDirectoryAndUploadFile(directory, filename, fileContent);
+
+        // WHEN
+        // THEN
+        testSuccessfulDelete(fileSystemName, directory, filename, inputFlowFileContent, inputFlowFileContent, DIRECTORY);
+    }
+
+    @Test
+    public void testDeleteEmptyDirectoryAsFolder() {
```

Comment:
It could simply be called `testDeleteEmptyDirectory` or does the `AsFolder` suffix has any extra meaning here?

Reply:
Yes, originally you could delete an empty folder if you provided it in the filename property. Now we have introduced a new property to chose between file and folder deletion. So that currently we can delete an empty directory as a FILE or as a FOLDER. I think I will rename the other test case, too. Or shall I add a comment somewhere about this? It's not trivial at all.

Reply:
I renamed the tests for
- testDeleteEmptyDirectoryWithFSTypeFile
- testDeleteEmptyDirectoryWithFSTypeDirectory
as an idea.